packages/@aws-cdk/aws-stepfunctions-tasks/lib/eventbridge/put-events.ts

```diff
@@ -0,0 +1,158 @@
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * An entry to be sent to EventBridge
+ * @see https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_PutEventsRequestEntry.html
+ */
+export interface IEventBridgePutEventsEntry {
+  /**
+   * JSON object that contains information about the event
+   * The service generating the event determines the content of this field
+   * Consists of a valid JSON string which can contain nested subobjects
+   * i.e. "{ \"instance-id\": \" i-1234567890abcdef0\", \"state\": \"terminated\" }"
+   */
+  readonly detail?: sfn.TaskInput;
+
+  /**
+   * Along with the source field help identify fields and values of detail field
+   * i.e. Events by CloudTrail have a "detail-type": "AWS API Call via CloudTrail",
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html
+   */
+  readonly detailType?: string;
+
+  /**
+   * The event bus the entry will be sent to.
+   * @default Send event to account's default event bus
+   */
+  readonly eventBus?: events.IEventBus;
+
+  /**
+   * JSON array containing ARNs of resources used in the event
+   * i.e. Auto Scaling events include ARNs for both instances and Auto Scaling groups
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html
+   * @default empty If the event does not include resource ARNs
+   */
+  readonly resources?: string[];
```

Comment:
Is this a useful thing for customers to specify? I can understand why a AWS service would want to supply the related ARNs but not why a user would want to instead of just putting everything in detail

Reply:
I assume there might be a scenario with nested state machines where the user would like to provide arns for later tasks.

Reply:
but why can't that be provided somewhere in the payload (the "detail" field). the only valuable thing I can see being populated here is the ARN of the state machine itself

Reply:
Fair enough, I can remove this field and customers can get the same access by using the detail field.